app/models/user.rb

```diff
@@ -0,0 +1,3 @@
+class User < ApplicationRecord
+  validates :username, presence: true, uniqueness: true
```

Comment:
User model has a relationship with with Reservation model
A user has many Reservations. To make things more convenient you should add the has_many relationship to the user model for Reservation